### PR TITLE
DDCE-4843 - update the logic for P11D and Forecast display

### PR DESCRIPTION
**Pull request by mywyau**
Backend change -> https://github.com/hmrc/tax-history/pull/170

### CODE REVIEW

**Comment by lmugnai** [comment]
@mywyau there is a bobby warning:

this:
addSbtPlugin("com.beautiful-scala" % "sbt-scalastyle" % "1.5.1")

can be changed to:
addSbtPlugin("org.scalastyle" % "scalastyle-sbt-plugin" % "1.0.0")

File: app/controllers/EmploymentDetailController.scala [hunk 1]
```diff
@@ -152,9 +152,9 @@ class EmploymentDetailController @Inject() (
   )(implicit hc: HeaderCarrier, request: Request[_]) = {
     val employment = dateUtils.formatEmploymentDates(empResponse.json.as[Employment])
     for {
-      payAndTax       <- getPayAndTax(nino, taxYear, employmentId)
-      companyBenefits <- getCompanyBenefits(nino, taxYear, employmentId)
-      incomeSource    <- getIncomeSource(nino, taxYear, employmentId)
+      payAndTax                             <- getPayAndTax(nino, taxYear, employmentId)
```
Comment (lmugnai) [inline app/controllers/EmploymentDetailController.scala:155]: Can we add type annotation here too?

File: app/controllers/EmploymentDetailController.scala [hunk 2]
```diff
@@ -152,9 +152,9 @@ class EmploymentDetailController @Inject() (
...
+      companyBenefits: List[CompanyBenefit] <- getCompanyBenefits(nino, taxYear, employmentId)
+      incomeSource                          <- getIncomeSource(nino, taxYear, employmentId)
```
Comment (lmugnai) [inline app/controllers/EmploymentDetailController.scala:157]: Can we add type annotation here too?

File: app/model/api/CompanyBenefit.scala [hunk 3]
```diff
@@ -25,9 +26,14 @@ case class CompanyBenefit(
   iabdType: String,
   amount: BigDecimal,
   source: Option[Int] = None,
+  captureDate: Option[String],
+  taxYear: TaxYear,
```
Comment (lmugnai) [inline app/model/api/CompanyBenefit.scala:30]: Do we need these in the FE model? I don't think they're used anywhere.

File: app/model/api/CompanyBenefit.scala [hunk 4]
```diff
@@ -25,9 +26,14 @@ case class CompanyBenefit(
...
   isForecastBenefit: Boolean
 )
 
 object CompanyBenefit {
+
+  implicit val formatTaxYear: OFormat[TaxYear] = Json.format[TaxYear]
```
Comment (lmugnai) [inline app/model/api/CompanyBenefit.scala:36]: Is this needed?

File: app/controllers/EmploymentDetailController.scala (hunk 1 above)
Comment (mywyau) [reply app/controllers/EmploymentDetailController.scala:155]: makes it easier to follow so yeah will do

File: app/model/api/CompanyBenefit.scala (hunk 4 above)
Comment (mywyau) [reply app/model/api/CompanyBenefit.scala:36]: only if we need to add it to the model.

**Pull request closed without merging**